crates/wasm-encoder/src/code.rs

```diff
@@ -140,8 +138,13 @@ impl Function {
         self
     }
 
-    /// Encode function to bytes
-    pub fn encode(&self, bytes: &mut Vec<u8>) {
+    /// Add raw bytes to this function's body including locals.
+    pub fn full_raw(data: Vec<u8>) -> Self {
+        let r = Self { bytes: data };
+        r
+    }
```

Comment:
I want to create a function by providing its full body. Right now, we could use the `raw` function, however, we need to instantiate by providing its local variables. With this method we set all, including the locals already encoded in the data.

Reply:
Right, but why do you want to do that? I don't think `wasm-mutate` should ever need to create a function from arbitrary bytes.

Reply:
Just to bypass those functions that wont be mutated. The other way is to completely parse the  `FunctionBody` and then construct the `Function` for wasm-encoder.

Reply:
Ahhh, I see, basically the equivalent of `RawSection` but at the function body granularity instead of section granularity. Yes, this makes perfect sense.

To support this use case, I think it will actually be better to add a method to `CodeSection` that takes a raw function as a `&[u8]` instead of constructing a `wasm_encoder::Function` with a `Vec<u8>`. This allows us to do only a single copy from the original input into the `wasm_encoder::Module` output, and we don't need to make an intermediate `Vec<u8>` copy.

Reply:
Gotcha, doing so then